File: app/src/main/java/konradrutkowski/com/tapapp/online/NetworkConnection.java
package konradrutkowski.com.tapapp.online;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;


public class NetworkConnection extends Thread {
    public static boolean isConnectingToInternet(Context ctx) {

        ConnectivityManager cm =
                (ConnectivityManager) ctx.getSystemService(Context.CONNECTIVITY_SERVICE);

        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        return activeNetwork != null &&
                activeNetwork.isConnectedOrConnecting();
    }
}

File: app/src/main/java/konradrutkowski/com/tapapp/online/NetworkConnection.kt
package konradrutkowski.com.tapapp.online

import android.content.Context
import android.net.ConnectivityManager


object NetworkConnection {

    fun isInternetReady(context: Context): Boolean {
        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val activeNetwork = connectivityManager.activeNetworkInfo

        return activeNetwork != null && activeNetwork.isConnectedOrConnecting
    }
}
